sdk/inc/azure/core/az_log.h

```diff
@@ -64,55 +61,62 @@ enum az_log_classification_core
 };
 
 /**
- * @brief Defines the signature of the callback function that application developers must write in
- * order to receive Azure SDK log messages.
+ * @brief Defines the signature of the callback function that application developers must provide to
+ * receive Azure SDK log messages.
  *
  * @param[in] classification The log message's #az_log_classification.
  * @param[in] message The log message.
  */
 typedef void (*az_log_message_fn)(az_log_classification classification, az_span message);
 
 /**
- * @brief Allows the application to specify which #az_log_classification types it is interested in
- * receiving.
+ * @brief Defines the signature of the callback function that application developers must provide
+ * which will be used to check whether a particular log classification should be logged.
  *
- * @param[in] classifications __[nullable]__ An array of #az_log_classification values, terminated
- * by #AZ_LOG_END_OF_LIST.
+ * @param[in] classification The log message's #az_log_classification.
  *
- * @details If no classifications are set (\p classifications is `NULL`), the application will
- * receive log messages for all #az_log_classification values.
- * @details If \p classifications is not `NULL`, it must point to an array of
- * #az_log_classification, terminated by #AZ_LOG_END_OF_LIST.
- * @details In contrast to \p classifications being `NULL`, \p classifications pointing to an empty
- * array (which still should be terminated by #AZ_LOG_END_OF_LIST), states that an application is
- * not interested in receiving any log messages.
+ * @return Whether or not a log message with the provided classification should be logged.
+ */
+typedef bool (*az_log_filter_fn)(az_log_classification classification);
+
+/**
+ * @brief Sets the functions that will be invoked to report an SDK log message.
+ *
+ * @param[in] log_message_callback __[nullable]__ A pointer to the function that will be invoked
+ * when the SDK reports a log message that should be logged according to the result of the
+ * #az_log_filter_fn provided to #az_log_set_classification_filter_callback(). If `NULL`, no
+ * function will be invoked.
  *
- * @warning Users must not change the \p classifications array elements, once it is passed to this
- * function. If \p classifications array is allocated on a stack, program behavior in multi-threaded
- * environment is undefined.
+ * @remarks By default, this is `NULL`, which means, no function is invoked.
  */
 #ifndef AZ_NO_LOGGING
-void az_log_set_classifications(az_log_classification const classifications[]);
+void az_log_set_message_callback(az_log_message_fn log_message_callback);
```

Comment:
I would leave it as `az_log_set_callback()`, reason being that this is the main line of thought when logging, so you think of it as "log callback", while "classification filter callback" is a specialization which you may, or may not provide.